File: src/dataset/cpp/space-age_exercism-pawelo091991/claude/ClaudeSonnet4_space_age_v3.cpp
#include "space_age.h"

space_age::space_age::space_age(double input) noexcept : time(input) {}

double space_age::space_age::calculate(double planet) const noexcept {
    return time / (earth * planet);
}

double space_age::space_age::seconds(void) const noexcept {
    return time;
}

double space_age::space_age::on_earth(void) const noexcept {
    return time / earth;
}

double space_age::space_age::on_mercury(void) const noexcept {
    return time / (earth * mercury);
}

double space_age::space_age::on_venus(void) const noexcept {
    return time / (earth * venus);
}

double space_age::space_age::on_mars(void) const noexcept {
    return time / (earth * mars);
}

double space_age::space_age::on_jupiter(void) const noexcept {
    return time / (earth * jupiter);
}

double space_age::space_age::on_saturn(void) const noexcept {
    return time / (earth * saturn);
}

double space_age::space_age::on_uranus(void) const noexcept {
    return time / (earth * uranus);
}

double space_age::space_age::on_neptune(void) const noexcept {
    return time / (earth * neptune);
}

File: src/dataset/python/space-age_exercism-ThomasZumsteg/claude/ClaudeSonnet4_space_age_v3.py
"""Calculates age on various planets"""

class SpaceAge(object):
    """Calculates age on various planets"""
    _earth_year = 31557600  # seconds
    _planet_ratios = {
        'earth': 1.0,
        'mercury': 0.2408467,
        'venus': 0.61519726,
        'mars': 1.8808158,
        'jupiter': 11.862615,
        'saturn': 29.447498,
        'uranus': 84.016846,
        'neptune': 164.79132
    }

    def __init__(self, seconds):
        """Stores age builds functions"""
        self.seconds = seconds
        self._cached_ages = {}

    def on_planet_gen(self, ratio_to_earth):
        """Returns a function that converts seconds into planet years"""
        def on_planet():
            """Converts seconds to planet years"""
            if ratio_to_earth not in self._cached_ages:
                self._cached_ages[ratio_to_earth] = round(
                    self.seconds / (self._earth_year * ratio_to_earth), 2
                )
            return self._cached_ages[ratio_to_earth]
        return on_planet

    def on_earth(self):
        return self.on_planet_gen(self._planet_ratios['earth'])()

    def on_mercury(self):
        return self.on_planet_gen(self._planet_ratios['mercury'])()

    def on_venus(self):
        return self.on_planet_gen(self._planet_ratios['venus'])()

    def on_mars(self):
        return self.on_planet_gen(self._planet_ratios['mars'])()

    def on_jupiter(self):
        return self.on_planet_gen(self._planet_ratios['jupiter'])()

    def on_saturn(self):
        return self.on_planet_gen(self._planet_ratios['saturn'])()

    def on_uranus(self):
        return self.on_planet_gen(self._planet_ratios['uranus'])()

    def on_neptune(self):
        return self.on_planet_gen(self._planet_ratios['neptune'])()